community/modules/scheduler/k8s-cluster/main.tf

```diff
@@ -0,0 +1,261 @@
+    }
+  }
+
+  pod_security_policy_config {
+    enabled = true
+  }
+
+  enable_shielded_nodes = true
+
+  cluster_autoscaling { # Auto provisioning of node-pools
+    enabled = false
+    # Recomended profile if we ever turn on
+    # autoscaling_profile = "OPTIMIZE_UTILIZATION"
+  }
+
+  datapath_provider = var.enable_dataplane_v2 ? "ADVANCED_DATAPATH" : "LEGACY_DATAPATH"
+
+  network_policy {
+    # Enabling NetworkPolicy for clusters with DatapathProvider=ADVANCED_DATAPATH
+    # is not allowed. Dataplane V2 will take care of network policy enforcement
+    # instead.
+    enabled = false
+    # GKE Dataplane V2 support. This must be set to PROVIDER_UNSPECIFIED in
+    # order to let the datapath_provider take effect.
+    # https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/issues/656#issuecomment-720398658
+    provider = "PROVIDER_UNSPECIFIED"
+  }
+
+  private_cluster_config {
+    enable_private_nodes    = var.enable_private_nodes
+    enable_private_endpoint = var.enable_private_endpoint
+    master_ipv4_cidr_block  = var.master_ipv4_cidr_block
+    master_global_access_config {
+      enabled = var.enable_master_global_access
+    }
+  }
+
+  ip_allocation_policy {
+    cluster_secondary_range_name  = var.pods_ip_range_name
+    services_secondary_range_name = var.services_ip_range_name
+  }
+
+  workload_identity_config {
+    workload_pool = "${var.project_id}.svc.id.goog"
+  }
+
+  dynamic "authenticator_groups_config" {
+    for_each = local.cluster_authenticator_security_group
+    content {
+      security_group = authenticator_groups_config.value.security_group
+    }
+  }
+
+  release_channel {
+    channel = var.release_channel
+  }
+
+  maintenance_policy {
+    daily_maintenance_window {
+      start_time = var.maintenance_start_time
+    }
+
+    dynamic "maintenance_exclusion" {
+      for_each = var.maintenance_exclusions
+      content {
+        exclusion_name = maintenance_exclusion.value.name
+        start_time     = maintenance_exclusion.value.start_time
+        end_time       = maintenance_exclusion.value.end_time
+        exclusion_options {
+          scope = maintenance_exclusion.value.exclusion_scope
+        }
+      }
+    }
+  }
+
+  addons_config {
+    # Istio is required if there is any pod-to-pod communication.
+    istio_config {
+      disabled = !var.enable_istio
+      auth     = var.istio_auth
+    }
+
+    gce_persistent_disk_csi_driver_config {
+      enabled = true
+    }
+  }
+
+  lifecycle {
+    # Ignore all changes to the default node pool. It's being removed after creation.
+    ignore_changes = [
+      node_config
+    ]
+  }
+
+  logging_service    = "logging.googleapis.com/kubernetes"
+  monitoring_service = "monitoring.googleapis.com/kubernetes"
+}
+
+# We define explict node pools, so that it can be modified without
+# having to destroy the entire cluster.
+resource "google_container_node_pool" "system-node-pools" {
+  provider = google-beta
```

Comment:
I am wondering if we could parametrize this provider. Does this need to be beta?

Reply:
then, it's ok as is.